src/Servers/Kestrel/Core/src/Internal/Http3/Http3Stream.cs

```diff
@@ -843,6 +850,25 @@ private static Task ProcessUnknownFrameAsync()
             {
                 throw new Http3StreamErrorException(CoreStrings.FormatHttp3DatagramStatusMismatch(_context.ClientPeerSettings.H3Datagram == 1, _context.ServerPeerSettings.H3Datagram == 1), Http3ErrorCode.SettingsError);
             }
+
+            if (string.Equals(HttpRequestHeaders.HeaderProtocol, WebTransportSession.WebTransportProtocolValue, StringComparison.Ordinal))
+            {
+                // indicate as webtransport request
+                _currentIHttpWebTransportFeature!.IsWebTransportRequest = true;
+
+                // TODO do as we go along and store as list in WebTransportSession or is that a memory issue?
+                // Also, should this be on a per webtransport session basis? If so, maybe I need a queue?
```

Comment:
I remember David mentioned that we try to avoid allocations, would this be worth the allocations or is doing it like this preferable?

Reply:
The lync APIs (Where, Select, etc) allocate a lot too. We limit their use in busy code paths. This can be replaced with a foreach loop.